subsys/bluetooth/host/adv.c

```diff
@@ -1777,12 +1777,25 @@ static void adv_timeout(struct k_work *work)
 int bt_le_per_adv_set_param(struct bt_le_ext_adv *adv,
 			    const struct bt_le_per_adv_param *param)
 {
+#if defined(CONFIG_BT_PER_ADV_RSP)
+	struct bt_hci_cp_le_set_per_adv_param_v2 *cp;
+#else
 	struct bt_hci_cp_le_set_per_adv_param *cp;
+#endif /* CONFIG_BT_PER_ADV_RSP */
```

Comment:
This must be an union. The check `BT_FEAT_LE_PAWR_ADVERTISER(bt_dev.le.features)` is runtime.

Reply:
Added a comment.